tasks/build/download_node_builds.js

```diff
@@ -1,84 +1,134 @@
-  let { resolve } = require('path');
-  let { pluck } = require('lodash');
-  let { createWriteStream } = require('fs');
-  let { createGunzip } = require('zlib');
-  let { Extract } = require('tar');
-  let { rename } = require('fs');
-  let wreck = require('wreck');
-
-  let platforms = grunt.config.get('platforms');
-  let activeDownloads = [];
-
-  let start = async (platform) => {
-    let finalDir = platform.nodeDir;
-    let downloadDir = `${finalDir}.temp`;
-
-    if (grunt.file.isDir(platform.nodeDir)) {
-      grunt.log.ok(`${platform.name} exists`);
-      return;
-    }
+import { map, fromNode, promisify } from 'bluebird';
+import { resolve, basename, dirname } from 'path';
+import { createReadStream, createWriteStream, writeFile } from 'fs';
+import { createGunzip } from 'zlib';
+import { Extract } from 'tar';
+import { rename } from 'fs';
+import checkHash  from 'check-hash';
+import wreck from 'wreck';
 
-    let resp = await fromNode(cb => {
-      let req = wreck.request('GET', platform.nodeUrl, null, function (err, resp) {
-        if (err) {
-          return cb(err);
-        }
+const wreckGet = promisify(wreck.get, wreck);
+const checkHashFromFile = promisify(checkHash.fromFile);
+const renameAsync = promisify(rename);
+const writeFileAsync = promisify(writeFile);
+
+module.exports = function (grunt) {
+  const platforms = grunt.config.get('platforms');
+  const downloadLimit = 3;
 
-        if (resp.statusCode !== 200) {
-          return cb(new Error(`${platform.nodeUrl} failed with a ${resp.statusCode} response`));
-        }
+  let shaSums = {};
+  const getShaSums = () => {
+    const nodeVersion = grunt.config.get('nodeVersion');
+    const shaSumsUri = `https://nodejs.org/dist/v${nodeVersion}/SHASUMS256.txt`;
 
-        return cb(null, resp);
+    return wreckGet(shaSumsUri).then(([resp, payload]) => {
+      if (resp.statusCode !== 200) {
+        throw new Error(`${shaSumsUri} failed with a ${resp.statusCode} response`);
+      }
+      payload.toString('utf8fix/7136 ').split('\n').forEach(line => {
+        const [sha, platform] = line.split('  ');
+        shaSums[platform] = sha;
       });
+      grunt.log.ok(`sha sums download complete`);
     });
+  };
 
-    // use an async iife to store promise for download
-    // then store platform in active downloads list
-    // which we will read from in the finish task
-    platform.downloadPromise = (async function () {
-      grunt.file.mkdir(downloadDir);
+  const checkShaSum = (platform) => {
+    const file = basename(platform.nodeUrl);
+    const path = resolve(platform.nodeDir, file);
+    const expected = {
+      hash: 'sha256',
+      expected: platform.win ? shaSums[basename(dirname(platform.nodeUrl)) + '/' + file] : shaSums[file]
+    };
+
+    return checkHashFromFile(path, expected).then(([passed, actual]) => {
+      if (passed) grunt.log.ok(`${platform.name} shasum verified`);
+      else grunt.log.error(`${platform.name} shasum check failed`);
```

Comment:
I'd recommend always using curly braces, omitting them is just asking for errors in the future